### ISSUE TITLE
Contact Matching Rule Fails When Case Mismatch

### ISSUE THREAD

**Issue by jesslopez-sf** [SalesforceFoundation/Volunteers-for-Salesforce#368]
When a customer modifies the Contact_Matching_Rule__c field in the Volunteers Settings Custom Setting and they don't use a specific case value for First Name /Last Name/ Email, matching fails and an existing Contact's data is overwritten and signed up for a Volunteer Job.

For example, this will not work:
`FirstName;LastName`

But this will work:
`Firstname;Lastname`

This is due to how we are checking for the presence of Firstname, Lastname, or Email within that Custom Setting on lines 773-798 of VOL_Shared_Code (https://github.com/SalesforceFoundation/Volunteers-for-Salesforce/blob/master/src/classes/VOL_SharedCode.cls#L773-L798)

Instead of using str.contains(), can we switch to str.containsIgnoreCase()? We've had a few reports of this recently. The documentation also shows examples using camel case for LastName, so this will fail if a customer copies/pastes.

Entering this in the Execute Anonymous window:
```
GW_Volunteers__Volunteers_Settings__c VolunteersSettings = GW_Volunteers__Volunteers_Settings__c.getInstance();
String cmr = VolunteersSettings.GW_Volunteers__Contact_Matching_Rule__c;
System.debug('cmr: ' + cmr);
System.debug('lower: ' + cmr.contains('Firstname'));
System.debug('upper: ' + cmr.contains('FirstName'));

```

Results in:
```
11:25:29:041 USER_DEBUG [3]|DEBUG|cmr: Firstname;Lastname;Email
11:25:29:041 USER_DEBUG [4]|DEBUG|lower: true
11:25:29:041 USER_DEBUG [5]|DEBUG|upper: false
```

**Comment by jesslopez-sf** [comment]
**lurch: add

**Comment by LurchTheButler** [comment]
Tracking <a href="https://foundation.my.salesforce.com/a2x80000007fOi3AAE" target="_blank">W-029131</a>

**Comment by mrbelvedere** [comment]
Included in beta release 3.110 (Beta 4)

**Comment by mrbelvedere** [comment]
Included in production release 3.110
